client/ayon_deadline/plugins/publish/global/collect_jobinfo.py

```diff
@@ -0,0 +1,331 @@
+                "chunk_size",
+                label="Frames Per Task",
+                default=default_values.get("chunk_size"),
+                decimals=0,
+                minimum=1,
+                maximum=1000
+            ),
+            NumberDef(
+                "priority",
+                label="Priority",
+                default=default_values.get("priority"),
+                decimals=0
+            ),
+            TextDef(
+                "department",
+                label="Department",
+                default=default_values.get("department")
+            ),
+            EnumDef(
+                "group",
+                label="Group",
+                default=default_values.get("group"),
+                items=cls.group_enum_values,
+            ),
+            EnumDef(
+                "limit_groups",
+                label="Limit Groups",
+                multiselection=True,
+                default=default_values.get("limit_groups"),
+                items=cls.limit_group_enum_values,
+            ),
+            EnumDef(
+                "primary_pool",
+                label="Primary pool",
+                default="none",
+                items=cls.pool_enum_values,
+            ),
+            EnumDef(
+                "secondary_pool",
+                label="Secondary pool",
+                default="none",
+                items=cls.pool_enum_values,
+            ),
+            EnumDef(
+                "machine_list",
+                label="Machine list",
+                multiselection=True,
+                default=default_values.get("machine_list"),
+                items=cls.machines_enum_values,
+            ),
+            BoolDef(
+                "machine_list_deny",
+                label="Machine List is a Deny",
+                default=default_values.get("machine_list_deny")
+            ),
+            TextDef(
+                "job_delay",
+                label="Delay job",
+                default=default_values["job_delay"],
+                tooltip="Delay job by specified timecode. Format: dd:hh:mm:ss",
+                placeholder="00:00:00:00"
+            )
+        ]
+
+        override_defs = []
+        for attr_def in attr_defs:
+            if attr_def.key not in overrides:
+                continue
+            override_defs.append(attr_def)
+
+        return override_defs
+
+    @classmethod
+    def register_create_context_callbacks(cls, create_context):
+        create_context.add_value_changed_callback(cls.on_values_changed)
+
+    @classmethod
+    def on_values_changed(cls, event):
+        for instance_change in event["changes"]:
+            instance = instance_change["instance"]
+            if not cls.instance_matches_plugin_families(instance):
+                continue
+
+            new_attrs = cls.get_attr_defs_for_instance(
+                event["create_context"], instance
+            )
+            instance.set_publish_plugin_attr_defs(cls.__name__, new_attrs)
```

Comment:
This has some major issues @kalisp @iLLiCiTiT 

1. This ALWAYS updates the attributes for all the instances that match the families on ANY attribute change. So we should be more selectively deciding on which attribute changes we should actually respond to here @kalisp ?

2. @iLLiCiTiT existing attribute definitions that do not change - do get swapped out completely to the publisher UI. Due to this setting the attribute defs again it means the user loses focus as they change those attributes. Preferably unchanged attributes should either NOT be re-assigned OR should not affect the user's focus in the publisher UI?

For example as soon as I type one character in a text field (e.g. the Department field) then the text field loses focus (likely because it gets recreated in the UI?).

Also, changing attributes (even unrelated to this plug-in) comes with a slight bit of lag because likely, yet again, we're constantly swapping out these widgets or recreating them? We may want to optimize the logic here first that so we do less redefining? It's not crazy slow, but it is noticable so we should really keep an eye out on how we can keep performance ok over time.

Reply:
I dont see behavior from 1). I have 2 instances of same family, if I update `chunk_size` for one, it will not change for the other (at least not in UI).

Reply:
It does not change the other no. I mean, any attribute change (even if not related to this set of attributes on the node) triggerrs the update of the attributes. Hence, go to a text field and try to enter a full word. As soon as I type one character, the field loses focus because the widget seems to get recreated. I can't type a word, or type multiple digits because it loses focus as I type because we're continuously 'replacing' the attribute definitions.

Reply:
Deservers issue in ayon-core.

Reply:
Issue created https://github.com/ynput/ayon-core/issues/1016 . But is not relevant for this PR anymore, resolving comment.